code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -0,0 +1,305 @@
+//Elite mining mobs
+/mob/living/simple_animal/hostile/asteroid/elite
+	name = "elite"
+	desc = "An elite monster, found in one of the strange glowing crevices on lavaland."
+	icon = 'icons/mob/lavaland/elite_lavaland_monsters.dmi'
+	faction = list("mining_elite")
+	robust_searching = TRUE
+	ranged_ignores_vision = TRUE
+	ranged = 1
+	obj_damage = 5
+	var/chosen_attack = 1
+	vision_range = 6
+	aggro_vision_range = 18
+	var/list/attack_action_types = list()
+	environment_smash = ENVIRONMENT_SMASH_NONE  //This is to prevent elites smashing up the mining station, we'll make sure we can smash minerals fine below.
+	harm_intent_damage = 0 //Punching elites gets you nowhere
+	var/obj/structure/elite_crevice/myparent = null
+	var/can_talk = 0
+	stat_attack = UNCONSCIOUS
+	sentience_type = SENTIENCE_BOSS
+	
+		
+//Gives player-controlled variants the ability to swap attacks
+/mob/living/simple_animal/hostile/asteroid/elite/Initialize(mapload)
+	. = ..()
+	for(var/action_type in attack_action_types)
+		var/datum/action/innate/elite_attack/attack_action = new action_type()
+		attack_action.Grant(src)
+
+//Prevents elites from attacking members of their faction (can't hurt themselves either) and lets them mine rock with an attack despite not being able to smash walls.
+/mob/living/simple_animal/hostile/asteroid/elite/AttackingTarget()
+	if(istype(target, /mob/living/simple_animal/hostile))
+		var/mob/living/simple_animal/hostile/M = target
+		if(M.faction == src.faction)
+			return FALSE
+	. = ..()
+	if(ismineralturf(target))
+		var/turf/closed/mineral/M = target
+		M.gets_drilled()
+		
+//Elites can't talk (normally)!
+/mob/living/simple_animal/hostile/asteroid/elite/say(message, bubble_type, var/list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE, forced = null)
+	if(can_talk == 1)
+		. = ..()
+		return
```

Comment:
if you're going to use a bool for the second return do it for the first too

Reply:
Resolved in upcoming fix.